src/normalization.jl

```diff
@@ -106,11 +106,35 @@ function imsharpen(
 
     image_equalized_gray = Gray.(image_equalized)
 
-    image_smoothed = imfilter(image_equalized_gray, Kernel.gaussian(smoothing_param))
+    return unsharp_mask(image_equalized_gray, smoothing_param, intensity)
+end
+
+"""
+    unsharp_mask(image_gray, smoothing_param, intensity)
+
+Apply unsharp masking on (equalized) grayscale ([0, `clapmax`]) image to enhance its sharpness.
+
+# Arguments
+- `image_gray: The input grayscale image, typically already equalized.
+- `smoothing_param::Int`: The pixel radius for Gaussian blurring (typically between 1 and 10).
+- `intensity: The amount of sharpening to apply. Higher values result in more pronounced sharpening.
+
+# Returns
+The sharpened grayscale image with values clipped between 0 and `clapmax`.
+"""
+function unsharp_mask(image_gray, smoothing_param, intensity, clampmax)
+    image_smoothed = imfilter(image_gray, Kernel.gaussian(smoothing_param))
+    clamp!(image_smoothed, 0.0, clampmax)
+    image_sharpened = image_gray * (1 + intensity) .- image_smoothed * intensity
+    clamp!(image_sharpened, 0.0, clampmax)
+    return round.(Int, image_sharpened)
+end
 
-    image_sharpened =
-        image_equalized_gray .* (1 + intensity) .+ image_smoothed .* (-intensity)
-    return min.(max.(image_sharpened, 0.0), 1.0)
+# For old workflow in final2020.m
+function unsharp_mask(image_gray, smoothing_param, intensity)
```

Comment:
Does this function need to keep the original docstring?

Reply:
I'll add a docstring and deprecate the method. Keeping it so old tests continue to pass.